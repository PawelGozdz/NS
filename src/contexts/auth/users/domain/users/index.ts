export * from './errors';
export * from './events';
export * from './user.aggregate-root';
export * from './user.aggregate-root.fixture';
export * from './user.snapshot';
export * from './users-command-repository.interface';
export * from './users-query-repository.interface';
