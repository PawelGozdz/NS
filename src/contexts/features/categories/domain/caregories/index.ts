export * from './category-command-repository.interface';
export * from './category-entity.fixture';
export * from './category-query-repository.interface';
export * from './category.entity';
export * from './errors';
export * from './events';
