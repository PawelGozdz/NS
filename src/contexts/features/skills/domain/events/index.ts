export * from './skill-created.event';
