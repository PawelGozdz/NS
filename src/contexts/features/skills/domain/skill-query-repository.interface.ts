import { ISkillsQueryParams } from '@app/core';

export type SkillInfo = {
  id: number;
  name: string;
  context: string;
  parentId: number | null;
  description: string | null;
};

export abstract class ISkillsQueryRepository {
  abstract getManyBy(query: ISkillsQueryParams): Promise<SkillInfo[]>;
}
