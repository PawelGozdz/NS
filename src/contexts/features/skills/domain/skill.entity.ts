export class Skill {
  id: number;

  name: string;

  description?: string | null;

  parentId?: number | null;

  context: string;

  constructor(props: Skill) {
    this.id = props.id;
    this.name = props.name;
    this.description = props.description ?? null;
    this.parentId = props.parentId ?? null;
    this.context = props.context;
  }

  public static create({
    id,
    name,
    description,
    parentId,
    context,
  }: {
    id: number;
    name: string;
    description?: string | null;
    parentId?: number | null;
    context: string;
  }): Skill {
    const skill = new Skill({
      id,
      name,
      description: description ?? null,
      parentId: parentId ?? null,
      context,
    });

    return skill;
  }
}

export type ISkillCreateData = {
  name: string;
  description?: string | null;
  parentId?: number | null;
  context: string;
};

export type ISkillUpdateData = {
  id: number;
  name?: string;
  description?: string | null;
  context?: string;
  parentId?: number | null;
};
