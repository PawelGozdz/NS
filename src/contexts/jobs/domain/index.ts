export * from './job-position';
export * from './job-user-profile';
export * from './jobs';
export * from './skill';
