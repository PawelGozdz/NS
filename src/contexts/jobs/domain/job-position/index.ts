export * from './errors';
export * from './events';
export * from './job-position-command-repository.interface';
export * from './job-position-entity.fixture';
export * from './job-position-query-repository.interface';
export * from './job-position.entity';
export * from './job-position.snapshot';
