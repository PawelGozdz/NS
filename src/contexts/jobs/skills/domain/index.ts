export * from './errors';
export * from './events';
export * from './skill-command-repository.interface';
export * from './skill-entity.fixture';
export * from './skill-query-repository.interface';
export * from './skill.entity';
