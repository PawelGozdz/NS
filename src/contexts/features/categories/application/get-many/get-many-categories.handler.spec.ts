import { createMock } from '@golevelup/ts-jest';
import { Test } from '@nestjs/testing';
import { PinoLogger } from 'nestjs-pino';

import { Actor } from '@app/core';
import { ActorType } from '@libs/common';

import { CategoryInfo, ICategoriesQueryRepository } from '../../domain';
import { GetManyCategoriesHandler } from './get-many-categories.handler';
import { GetManyCategoriesQuery } from './get-many-categories.query';

describe('GetManyCategoriesHandler', () => {
  let handler: GetManyCategoriesHandler;
  let categoryQueryRepositoryMock: jest.Mocked<ICategoriesQueryRepository>;

  beforeEach(async () => {
    categoryQueryRepositoryMock = createMock<ICategoriesQueryRepository>();

    const app = await Test.createTestingModule({
      providers: [
        GetManyCategoriesHandler,
        {
          provide: ICategoriesQueryRepository,
          useValue: categoryQueryRepositoryMock,
        },
        {
          provide: PinoLogger,
          useValue: createMock<PinoLogger>(),
        },
      ],
    }).compile();

    handler = app.get(GetManyCategoriesHandler);
  });

  const actor = Actor.create(ActorType.GUEST, GetManyCategoriesHandler.name, 'c8aa6154-dba2-466c-8858-64c755e71ff1');

  const query: GetManyCategoriesQuery = {
    _filter: {
      name: 'it',
    },
    actor,
  };

  const categories: CategoryInfo[] = [
    {
      id: 2,
      name: 'name',
      description: 'description',
      parentId: 1,
    },
  ];

  it('should get many categories', async () => {
    // Arrange
    categoryQueryRepositoryMock.getManyBy.mockResolvedValueOnce(categories);

    // Act
    const result = await handler.execute(query);

    // Assert
    expect(result.length).toBe(1);
    expect(result).toMatchSnapshot();
  });
});
