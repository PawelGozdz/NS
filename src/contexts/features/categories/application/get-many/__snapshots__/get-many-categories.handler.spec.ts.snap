// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetManyCategoriesHandler should get many categories 1`] = `
[
  {
    "ctx": "ctx",
    "description": "description",
    "id": 2,
    "name": "name",
    "parentId": 1,
  },
]
`;
